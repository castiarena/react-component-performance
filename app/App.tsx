import React from 'react';
import { Switch, Route } from 'react-router-dom';
import styled, { createGlobalStyle, ThemeProvider } from 'styled-components';
import reset from 'styled-reset';
import customTheme from './theme';
import Test from './pages/test/Test';
import Performance from './pages/performance/Performance';
import Header from './components/Header';
import media from './components/Layout/MediaQueries';
import LazyLoad from './components/LazyLoad';

const MainWrapper = styled.div`
    font-family: 'Manrope', sans-serif;
    max-width: 1280px;
    margin: 0 auto;
    ${media.desktop}{
        padding: ${({ theme }) => theme.spacing.medium};
    }
    ${media.mobile}{
        padding: ${({ theme }) => `${theme.spacing.tiny} ${theme.spacing.tiny} ${theme.spacing.large}`} ;
    }
`;

const GlobalStyles = createGlobalStyle`
    ${reset};
    * {
        box-sizing: border-box;
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
    }
    html, body {
      background-color: #FBFCFF;
    }
`;

const App: React.FC = () => (
    <ThemeProvider theme={customTheme}>
        <MainWrapper>
            <GlobalStyles />
            <Header routes={[{
                to: '/',
                name: 'Home',
            }, {
                to: '/test',
                name: 'Test',
            }, {
                to: '/performance',
                name: 'Performance',
            }]}
            />
            <div>
                <Switch>
                    <Route exact path="/">
                        <LazyLoad importComponent={() => import('./pages/home/Home')} />
                    </Route>
                    <Route path="/test">
                        <LazyLoad importComponent={() => import('./pages/test/Test')} />
                    </Route>
                    <Route path="/performance">
                        <LazyLoad importComponent={() => import('./pages/performance/Performance')} />
                    </Route>
                </Switch>
            </div>
        </MainWrapper>
    </ThemeProvider>
);

export default App;
