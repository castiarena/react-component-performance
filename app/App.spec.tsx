import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import App from './App';

describe('App component test suite', () => {
    it('smoke test, snapshot only ', () => {
        const { asFragment } = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    });
});
