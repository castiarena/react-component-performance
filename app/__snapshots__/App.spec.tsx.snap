// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test suite smoke test, snapshot only  1`] = `
<DocumentFragment>
  .c1 {
  color: white;
  border: none;
  background-color: green;
  border-radius: 12px;
}

.c0 {
  font-family: 'Manrope',sans-serif;
}

<div
    class="c0"
  >
    <div>
      <h1>
        Hello world
      </h1>
      <button
        class="c1"
      >
        Button
      </button>
    </div>
  </div>
</DocumentFragment>
`;
